### PR TITLE
Prepare the UG documentation for Row drag and drop

### PR DESCRIPTION
**Pull request by ShaliniSF79**
- Prepared the UG documentation for row drag and drop

### CODE REVIEW

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+ * Group positions can be reordered using drag and drop. However, no groups can be added within other groups.
+ * Data rows can be reordered within the same group or moved to other groups as well.
+
+N> Reordering changes are made only in the `SfDataGrid.View`, and not in the underlying data. Thus, the changes will be reverted when performing sorting, grouping, or any other operation refreshing the view. Reordering changes in the underlying data can be achieved by handling a `QueryRowDragging` event in the sample side as explained below in [Reordering underlying data](# Reordering underlying data ).
+
+N> Now you can drag and drop the grid rows using your mouse in the UWP (Desktop) platform.
+
+## Row drag-and-drop template
+
+The data grid allows you to load desired content when performing row drag-and-drop operations using the `SfDataGrid.RowDragDropTemplate`. The template can be defined either in code or XAML.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:45]: The `SfDataGrid` allows you to load specific content during row drag-and-drop operations using the `SfDataGrid.RowDragDropTemplate`. This template can be defined either in code or XAML.

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+N> Now you can drag and drop the grid rows using your mouse in the UWP (Desktop) platform.
+
+## Row drag-and-drop template
+
+The data grid allows you to load desired content when performing row drag-and-drop operations using the `SfDataGrid.RowDragDropTemplate`. The template can be defined either in code or XAML.  
+
+## Default template
+
+The default template will be used if no other template is explicitly assigned for row drag-and-drop operations.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:49]: The default template will be used for row drag-and-drop operations unless another template is explicitly assigned.

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+            </Grid>
+        </DataTemplate>
+    </syncfusion:SfDataGrid.RowDragDropTemplate>
+</syncfusion:SfDataGrid>
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Default drag and drop" src="Images\row-drag-and-drop\LabelTemp.gif" />
+
+N> Currently, the row drag-and-drop features cannot be used if different row sets with varying heights are specified using the `QueryRowHeight` event.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:78]: Remove the note. This is not applicable.

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+The `QueryRowDragging` event provides the following properties in the `QueryRowDraggingEventArgs`:
+
+ * `From`: Returns the index of the row that is currently being dragged.
+ * `To`: Returns the index of the row where you attempt to drop the dragged item.
+ * `Position`: Returns the current x and y coordinates of the RowDragView.
+ * `DraggingAction`: Returns the details of row dragging as `DataGridDragAction`.
+ * `RowData`: Returns the underlying data associated with the row that was dragged.
+ * `CurrentRowData`: Returns the row data corresponding to the current position of the row drag view.
+ * `CanAutoScroll`: Returns whether auto-scrolling should occur when the row drag view reaches the top or bottom of the `SfDataGrid`.
+ * `Cancel`: Returns a Boolean value that indicates whether to cancel the event.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:93]: The `QueryRowDragging` event is triggered when a row begins to be dragged and continues to fire until the dragging ends. By handling the `SfDataGrid.QueryRowDragging` event, you can also cancel the dragging of a specific row.

The `QueryRowDragging` event provides the following properties through the `QueryRowDraggingEventArgs`:

 * `From`: Returns the index of the row that is currently being dragged.
 * `To`: Returns the index of the row where you are attempting to drop the dragged item.
 * `Position`: Returns the current x and y coordinates of the `RowDragView`.
 * `DraggingAction`: Returns details of the row dragging as a `DataGridDragAction`.
 * `RowData`: Returns the underlying data associated with the dragged row.
 * `CurrentRowData`: Returns the data corresponding to the current position of the `RowDragView`.
 * `CanAutoScroll`: Returns a value indicating whether auto-scrolling should occur when the `RowDragView` reaches the top or bottom of the `SfDataGrid`.
 * `Cancel`: Returns a Boolean value that indicates whether the event should be canceled.

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+{
+   e.CanAutoScroll = true;
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+
+## RowDragAndDrop Style
+### RowDraggingIndicatorLineColor
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:338]: Customization of draw view

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+
+The `RowDraggingIndicatorLineColor` property in the `SfDataGrid` is used to customize the color of the line that appears during a drag-and-drop operation when you move rows within the dataGrid. This line serves as a visual indicator, showing where the row will be dropped if the user releases the drag at that point.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:340]: The row view’s background color, text color, and indicator color can be customized using the RowDragViewBackgroundColor, RowDragViewTextColor, and RowDraggingIndicatorLineColor properties.

// Remove the three topics and provide only one code snippet for all the properties here and provide image also.

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+description: Learn here all about Row Drag and Drop support in Syncfusion MAUI DataGrid (SfDataGrid) control and more here.
+platform: MAUI
+control: SfDataGrid
+documentation: ug
+keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .net maui datagrid, .net maui grid, .net grid maui, .net maui row drag and drop, maui row drag and drop
+---
+
+# Row Drag and Drop in MAUI DataGrid (SfDataGrid)
+
+The SfDataGrid allows you to drag and drop a row by setting the `SfDataGrid.AllowDraggingRow` property to `true`. A customizable row drag and drop template is displayed while a row is being dragged.  Based on the requirements, drag and drop operations can be handled by using the `SfDataGrid.QueryRowDragging` event.
```
[the remark this answers is not in the document]
Comment (ShaliniSF79) [reply MAUI/DataGrid/row-drag-and-drop.md:13]: Updated

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 8]
```diff
@@ -0,0 +1,402 @@
...
+
+The SfDataGrid supports drag and drop operations for both data rows and groups in the following scenarios:
+
+ * Records can be reordered to any position using auto-scrolling.
+ * Group positions can be reordered using drag and drop. However, no groups can be added within other groups.
+ * Data rows can be reordered within the same group or moved to other groups as well.
+
+N> Reordering changes are made only in the `SfDataGrid.View`, and not in the underlying data. Thus, the changes will be reverted when performing sorting, grouping, or any other operation refreshing the view. Reordering changes in the underlying data can be achieved by handling a `QueryRowDragging` event in the sample side as explained below in [Reordering underlying data](# Reordering underlying data ).
+
+N> Now you can drag and drop the grid rows using your mouse in the UWP (Desktop) platform.
```
[the remark this answers is not in the document]
Comment (ShaliniSF79) [reply MAUI/DataGrid/row-drag-and-drop.md:41]: Removed

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 9]
```diff
@@ -0,0 +1,402 @@
...
+
+## Dragging scenarios
+
+The SfDataGrid supports drag and drop operations for both data rows and groups in the following scenarios:
+
+ * Records can be reordered to any position using auto-scrolling.
+ * Group positions can be reordered using drag and drop. However, no groups can be added within other groups.
+ * Data rows can be reordered within the same group or moved to other groups as well.
+
+N> Reordering changes are made only in the `SfDataGrid.View`, and not in the underlying data. Thus, the changes will be reverted when performing sorting, grouping, or any other operation refreshing the view. Reordering changes in the underlying data can be achieved by handling a `QueryRowDragging` event in the sample side as explained below in [Reordering underlying data](# Reordering underlying data ).
```
[the remark this answers is not in the document]
Comment (ShaliniSF79) [reply MAUI/DataGrid/row-drag-and-drop.md:39]: Changed

File: MAUI/DataGrid/row-drag-and-drop.md (hunk 1 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/row-drag-and-drop.md:45]: Changed

File: MAUI/DataGrid/row-drag-and-drop.md (hunk 2 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/row-drag-and-drop.md:49]: Updated

File: MAUI/DataGrid/row-drag-and-drop.md (hunk 3 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/row-drag-and-drop.md:78]: Removed

File: MAUI/DataGrid/row-drag-and-drop.md (hunk 4 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/row-drag-and-drop.md:93]: Updated

File: MAUI/DataGrid/row-drag-and-drop.md (hunk 5 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/row-drag-and-drop.md:338]: Changed

File: MAUI/DataGrid/row-drag-and-drop.md (hunk 6 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/row-drag-and-drop.md:340]: Removed and updated

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+
+<img alt="DataGrid row drag and drop" src="Images\row-drag-drop\maui-datagrid-dragging.gif" width="404"/>
+
+## Dragging scenarios
+
+The SfDataGrid supports drag-and-drop operations for both data rows and groups in the following scenarios:
+
+Records can be reordered to any position using auto-scrolling.
+Group positions can be reordered using drag and drop; however, groups cannot be nested within other groups.
+Data rows can be reordered within the same group or moved to different groups.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:37]: Provide bullet points to the above points.

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 11]
```diff
@@ -0,0 +1,360 @@
...
+
+N> Reordering changes are made only in the `SfDataGrid.View`, not in the underlying data. As a result, these changes will be reverted when performing sorting, grouping, or any other operation that refreshes the view. To persist reordering changes in the underlying data, handle the `QueryRowDragging` event as explained below in [Reordering underlying data](#Reordering underlying data).
+
+## Row drag-and-drop template
+
+The SfDataGrid allows you to load specific content during row drag-and-drop operations using the `SfDataGrid.RowDragDropTemplate`. This template can be defined either in code or XAML.
+
+## Default template
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:45]: It should be heading level 3.

      ### Default template

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+
+The default template will be used for row drag-and-drop operations unless another template is explicitly assigned.
+
+<img alt="DataGrid row drag and drop" src="Images\row-drag-drop\maui-datagrid-dragging.gif" width="404"/>
+
+## Customizing row drag-and-drop template
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:51]: It should be heading level 3.

      ### Customizing row drag-and-drop template

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 13]
```diff
@@ -0,0 +1,360 @@
...
+        <syncfusion:SfDataGrid.RowDragDropTemplate>
+        <DataTemplate>
+            <Grid BackgroundColor="LightBlue">
+                <Label Text="Drag View"       
+                       TextColor="Black"           
+                       HorizontalTextAlignment="Center" 
+                       VerticalTextAlignment="Center"/>
+            </Grid>
+        </DataTemplate>
+    </syncfusion:SfDataGrid.RowDragDropTemplate>
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:69]: Provide full code snippets along with the SfDataGrid tag.

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+private async void DataGrid_QueryRowDragging(object? sender, Syncfusion.Maui.DataGrid.DataGridQueryRowDraggingEventArgs e)
+{
+   e.CanAutoScroll = true;
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+
+## Customization of draw view
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:334]: Customization of drag view

File: MAUI/DataGrid/row-drag-and-drop.md [hunk 15]
```diff
@@ -0,0 +1,360 @@
...
+
+The row view’s background color, text color, and indicator color can be customized using the `RowDragViewBackgroundColor`, `RowDragViewTextColor`, and `RowDraggingIndicatorLineColor` properties.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/row-drag-and-drop.md:336]: The row drag view’s background color, text color, and indicator color can be customized using the `RowDragViewBackgroundColor`, `RowDragViewTextColor`, and `RowDraggingIndicatorLineColor` properties.

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]
Need to revamp appearance image.

**Review by ashok-kuvaraja — APPROVED** [review]
The code changes are fine

**Pull request merged**